components/services/bat_ads/bat_ads_client_mojo_bridge.cc

```diff
@@ -290,59 +290,84 @@ void BatAdsClientMojoBridge::Log(const char* file,
 }
 
 absl::optional<base::Value> BatAdsClientMojoBridge::GetProfilePref(
-    const std::string& path) const {
+    const std::string& path) {
   if (!bat_ads_client_associated_receiver_.is_bound()) {
-    return absl::nullopt;
+    return GetCachedProfilePrefValue(path);
   }
 
   absl::optional<base::Value> value;
-  bat_ads_client_associated_receiver_->GetProfilePref(path, &value);
+  if (!bat_ads_client_associated_receiver_->GetProfilePref(path, &value)) {
+    return GetCachedProfilePrefValue(path);
+  }
+
+  if (value) {
+    cached_profile_prefs_[path] = value->Clone();
+  } else {
+    cached_profile_prefs_.erase(path);
+  }
+
   return value;
 }
 
 void BatAdsClientMojoBridge::SetProfilePref(const std::string& path,
                                             base::Value value) {
+  cached_profile_prefs_[path] = value.Clone();
   if (bat_ads_client_associated_receiver_.is_bound()) {
     bat_ads_client_associated_receiver_->SetProfilePref(path, std::move(value));
   }
 }
 
 void BatAdsClientMojoBridge::ClearProfilePref(const std::string& path) {
+  cached_profile_prefs_.erase(path);
   if (bat_ads_client_associated_receiver_.is_bound()) {
     bat_ads_client_associated_receiver_->ClearProfilePref(path);
   }
 }
 
 bool BatAdsClientMojoBridge::HasProfilePrefPath(const std::string& path) const {
   if (!bat_ads_client_associated_receiver_.is_bound()) {
-    return false;
+    return cached_profile_prefs_.contains(path);
   }
 
   bool value = false;
-  bat_ads_client_associated_receiver_->HasProfilePrefPath(path, &value);
+  if (!bat_ads_client_associated_receiver_->HasProfilePrefPath(path, &value)) {
+    return cached_profile_prefs_.contains(path);
+  }
+
   return value;
 }
 
 absl::optional<base::Value> BatAdsClientMojoBridge::GetLocalStatePref(
-    const std::string& path) const {
+    const std::string& path) {
   if (!bat_ads_client_associated_receiver_.is_bound()) {
-    return absl::nullopt;
+    return GetCachedLocalStatePrefValue(path);
   }
 
   absl::optional<base::Value> value;
-  bat_ads_client_associated_receiver_->GetLocalStatePref(path, &value);
+  if (!bat_ads_client_associated_receiver_->GetLocalStatePref(path, &value)) {
+    return GetCachedLocalStatePrefValue(path);
+  }
+
+  if (value) {
```

Comment:
See comment for `GetProfilePref`

Reply:
nullopt here is also possible based on our Mojom interface definition:

```
GetLocalStatePref(string path) => (mojo_base.mojom.Value? value);
```